peer/brontide.go

```diff
@@ -1983,13 +1985,44 @@ func (p *Brontide) pingHandler() {
 	defer pingTicker.Stop()
 
 	// TODO(roasbeef): make dynamic in order to create fake cover traffic
-	const numPingBytes = 16
+	const numPongBytes = 16
 
+	blockEpochs, err := p.cfg.ChainNotifier.RegisterBlockEpochNtfn(nil)
+	if err != nil {
+		peerLog.Errorf("unable to establish block epoch "+
+			"subscription: %v", err)
+	}
+
+	var (
+		pingPayload [wire.MaxBlockHeaderPayload]byte
+		blockHeader *wire.BlockHeader
+	)
 out:
 	for {
 		select {
+		// Each time a new block comes in, we'll copy the raw header
+		// contents over to our ping payload declared above. Over time,
+		// we'll use this to disseminate the latest block header
+		// between all our peers, which can later be used to
+		// cross-check our own view of the network to mitigate various
+		// types of eclipse attacks.
+		case epoch := <-blockEpochs.Epochs:
+			blockHeader = epoch.BlockHeader
+			headerBuf := bytes.NewBuffer(pingPayload[0:0])
+			err := blockHeader.Serialize(headerBuf)
+			if err != nil {
+				peerLog.Errorf("unable to encode header: %v",
+					err)
+			}
+
 		case <-pingTicker.C:
-			p.queueMsg(lnwire.NewPing(numPingBytes), nil)
+
+			pingMsg := &lnwire.Ping{
+				NumPongBytes: numPongBytes,
+				PaddingBytes: pingPayload[:],
```

Comment:
would this not require a spec change? I see currently in the spec the field is marked as `ignored` but what if another impl also decides to use the field for a different type of data? would that not cause an interop issue?

Reply:
another question is should we not also pad the `pingPayload` randomly to "allow obfuscation of traffic patterns" as the spec says?

Reply:
No spec change is needed here IMO, it's just a field that lets a sender add some additional data to the ping payload. The measure was carved out to eventually allow for more advanced anti-traffic analysis usage. IMO we'd need to actually dig in and do some research+analysis before attempting to claim that anything we do here re random values actually has an impact re mitigating active traffic analysis by a network observer. 

> but what if another impl also decides to use the field for a different type of data? would that not cause an interop issue?

Don't see an issue here, since it'll either decode as a block header or not. It's also easy to pattern match (look at the length or certain magic values used in headers/encoding). Eventually if this catches on, it may make sense to add a node level feature bit so nodes can seek out other nodes that send block headers in the ping. 

> another question is should we not also pad the pingPayload randomly to "allow obfuscation of traffic patterns" as the spec says?

We already use a fixed value here (16 before this PR), see above re what (IMO) we'd need to do before claiming this has any sort of privacy enhancing capabilities.